scripts/openmc-make-depletion-chain-casl

```diff
@@ -0,0 +1,240 @@
+        endf_dir = '.'
+    elif 'OPENMC_ENDF_DATA' in os.environ:
+        endf_dir = os.environ['OPENMC_ENDF_DATA']
+    else:
+        for url in URLS:
+            basename = download(url)
+            with ZipFile(basename, 'r') as zf:
+                print('Extracting {}...'.format(basename))
+                zf.extractall()
+        endf_dir = '.'
+
+    decay_files = glob.glob(os.path.join(endf_dir, 'decay', '*.endf'))
+    fpy_files = glob.glob(os.path.join(endf_dir, 'nfy', '*.endf'))
+    neutron_files = glob.glob(os.path.join(endf_dir, 'neutrons', '*.endf'))
+
+    # Create a Chain
+    chain = openmc.deplete.Chain()
+
+    print('Reading ENDF nuclear data from "{}"...'.format(os.path.abspath(endf_dir)))
+
+    # Create dictionary mapping target to filename
+    print('Processing neutron sub-library files...')
+    reactions = {}
+    for f in neutron_files:
+        evaluation = openmc.data.endf.Evaluation(f)
+        name = evaluation.gnd_name
+        if name in CASL_CHAIN:
+            reactions[name] = {}
+            for mf, mt, nc, mod in evaluation.reaction_list:
+                if mf == 3:
+                    file_obj = StringIO(evaluation.section[3, mt])
+                    openmc.data.endf.get_head_record(file_obj)
+                    q_value = openmc.data.endf.get_cont_record(file_obj)[1]
+                    reactions[name][mt] = q_value
+
+    # Determine what decay and FPY nuclides are available
+    print('Processing decay sub-library files...')
+    decay_data = {}
+    for f in decay_files:
+        data = openmc.data.Decay(f)
+        name = data.nuclide['name']
+        if name in CASL_CHAIN:
+            decay_data[name] = data
+
+    for name in CASL_CHAIN:
+        if name not in decay_data:
+            print('WARNING: {} has no decay data!'.format(name))
+
+    print('Processing fission product yield sub-library files...')
+    fpy_data = {}
+    for f in fpy_files:
+        data = openmc.data.FissionProductYields(f)
+        name = data.nuclide['name']
+        if name in CASL_CHAIN:
+            fpy_data[name] = data
+
+    print('Creating depletion_chain...')
+    missing_daughter = []
+    missing_rx_product = []
+    missing_fpy = []
+
+    for idx, parent in enumerate(sorted(decay_data, key=openmc.data.zam)):
+        data = decay_data[parent]
+
+        nuclide = Nuclide()
+        nuclide.name = parent
+
+        chain.nuclides.append(nuclide)
+        chain.nuclide_dict[parent] = idx
+
+        if not CASL_CHAIN[parent][0] and \
+           not data.nuclide['stable'] and data.half_life.nominal_value != 0.0:
+            nuclide.half_life = data.half_life.nominal_value
+            nuclide.decay_energy = sum(E.nominal_value for E in
+                                       data.average_energies.values())
+            sum_br = 0.0
+            for i, mode in enumerate(data.modes):
+                type_ = ','.join(mode.modes)
```

Comment:
Nit: Would recommend `decay_mode`, `decay_mode_type`, or `mode_type` for variable identifier rather than `type_`.  Makes it slightly easier to read as the parameter into the `DecayTuple` constructor [1].

[1] https://openmc.readthedocs.io/en/latest/pythonapi/generated/openmc.deplete.DecayTuple.html